usr/src/uts/common/io/scsi/adapters/smartpqi/smartpqi_util.c

```diff
@@ -0,0 +1,947 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2018 Nexenta Systems, Inc.
+ */
+
+/*
+ * Utility routines that have common usage throughout the driver.
+ */
+#include <smartpqi.h>
+
+/* ---- Forward declarations for support/utility functions ---- */
+static void reinit_io(pqi_io_request_t *io);
+static char *cmd_state_str(pqi_cmd_state_t state);
+static void dump_raid(pqi_state_t s, void *v, pqi_index_t idx);
+static void dump_aio(void *v);
+static void show_error_detail(pqi_state_t s);
+static void cmd_finish_task(void *v);
+
+/*
+ * []------------------------------------------------------------------[]
+ * | Entry points for this file						|
+ * []------------------------------------------------------------------[]
+ */
+
+int
+pqi_is_offline(pqi_state_t s)
```

Comment:
Should return a boolean.

I also got the impression this was used inconsistently in the code.

Reply:
Normally it's polite to prefix your *external* symbol names with the name of your driver.